var/spack/repos/builtin/packages/py-melissa/package.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyMelissa(PythonPackage):
+    """Melissa is a file-avoiding, adaptive, fault-tolerant and elastic
+    framework, to run large-scale sensitivity analysis or deep-surrogate
+    training on supercomputers.
+    This package builds the launcher and server modules.
+    """
+
+    homepage = "https://gitlab.inria.fr/melissa/melissa"
+    git = "https://gitlab.inria.fr/melissa/melissa.git"
+    maintainers("robcaulk", "mschouler", "raffino")
+
+    version("develop", branch="develop")
+
+    # define variants (DL, DL-Torch, DL-Tensorflow)
+    variant(
+        "MELISSA_DL",
+        default=False,
+        description="Install all Deep Learning requirements (Pytorch and TensorFlow)",
+    )
+    variant(
+        "MELISSA_TORCH",
+        default=False,
+        description="Install Deep Learning requirements with Pytorch only",
+    )
+    variant(
+        "MELISSA_TF",
+        default=False,
+        description="Install Deep Learning requirements with TensorFlow only",
+    )
+
+    depends_on("python@3.8.0:", type=("build", "run"))
+    depends_on("py-setuptools", type=("build"))
+    # requirements.txt (SA - note that plotext is not available with spack)
```

Comment:
Assuming that everything in requirements.txt is actually required, you'll need to add a package for plotext

Reply:
`plotext` is used to monitor the progress of a Melissa study but it is not a vital component of the framework. Since we are not involved in the development of `plotext`, we would rather not add a package for it. Instead we will make it a dev requirement.